nucypher/characters/lawful.py

```diff
@@ -1049,21 +1064,42 @@ def __init__(self,
                          decentralized_identity_evidence=decentralized_identity_evidence,
                          )
 
+        #
+        # Startup Services  # TODO: PR #1462
+        #
+
         if start_learning_now:
             self.start_learning_loop(now=True)
 
-        #
-        # Logging / Updating
-        #
         if is_me:
+
+            # Initial Fleet State
             self.known_nodes.record_fleet_state(additional_nodes_to_track=[self])
+
+            # Arrangement Pruning
+            self._arrangement_pruning_task = LoopingCall(f=self.__prune_arrangements)
+            self.__pruning_task = None   # TODO: Move to ursula.run awaiting PR #1462
+            self.__pruning_task = self._arrangement_pruning_task.start(interval=self._pruning_interval)
+
             message = "THIS IS YOU: {}: {}".format(self.__class__.__name__, self)
             self.log.info(message)
             self.log.info(self.banner.format(self.nickname))
+
         else:
             message = "Initialized Stranger {} | {}".format(self.__class__.__name__, self)
             self.log.debug(message)
 
+    def __prune_arrangements(self) -> None:
+        """Deletes all expired arrangements and kfrags in the datastore."""
+        now = datetime.fromtimestamp(self._arrangement_pruning_task.clock.seconds())
+        try:
+            result = self.datastore.del_expired_policy_arrangements(now=now)
+        except OperationalError:
+            self.log.warn(f"Failed to prune policy arrangements; DB session rolled back.")
+        else:
+            self.log.debug(f"Pruned {result} policy arrangements.")
```

Comment:
To reduce the "noise" in logs:
```suggestion
            if result > 0:
                self.log.debug(f"Pruned {result} policy arrangements.")
```

Reply:
:heavy_check_mark: